src/analysis/inner_product_space/gram_schmidt_ortho.lean

```diff
@@ -103,90 +103,86 @@ open submodule set order
 end
 
-/-- If the input of the first `n + 1` vectors of `gram_schmidt` are linearly independent,
-then the output of the first `n + 1` vectors are non-zero. -/
+/-- If the input of the first `n` vectors of `gram_schmidt` are linearly independent,
+then the output of the first `n` vectors are non-zero. -/
 lemma gram_schmidt_ne_zero (f : ℕ → E) (n : ℕ)
-  (h₀ : linear_independent 𝕜 (f ∘ (coe : fin n.succ → ℕ))) :
-    gram_schmidt 𝕜 f n ≠ 0 :=
+  (h₀ : linear_independent 𝕜 (f ∘ (coe : fin n → ℕ))) :
+    ∀ i, i < n → gram_schmidt 𝕜 f i ≠ 0 :=
 begin
   induction n with n hn,
-  { intro h,
-    simp only [gram_schmidt_zero, ne.def] at h,
-    exact linear_independent.ne_zero 0 h₀ (by simp only [function.comp_app, fin.coe_zero, h]), },
-  { by_contra h₁,
-    rw nat.succ_eq_add_one at hn h₀ h₁,
-    have h₂ := gram_schmidt_def' 𝕜 f n.succ,
+  { intros, linarith },
+  { intros i hi h₁,
+    rw nat.succ_eq_add_one at hi,
+    have h₂ := gram_schmidt_def' 𝕜 f i,
     simp only [nat.succ_eq_add_one, h₁, orthogonal_projection_singleton, zero_add] at h₂,
-    have h₃ : f (n + 1) ∈ span 𝕜 (f '' Iic n),
-    { rw [h₂, ← span_gram_schmidt 𝕜 f n],
+    have h₃ : f i ∈ span 𝕜 (f '' Iio i),
+    { rw [h₂, ← span_gram_schmidt 𝕜 f i],
       apply submodule.sum_mem _ _,
       simp_intros a ha only [finset.mem_range],
       apply submodule.smul_mem _ _ _,
       refine subset_span _,
-      simp only [mem_image, mem_Iic],
+      simp only [mem_image, mem_Iio],
       exact ⟨a, by linarith, by refl⟩, },
-    change linear_independent 𝕜 (f ∘ (coe : fin (n + 2) → ℕ)) at h₀,
-    have h₄ : ((n + 1) : fin (n + 2)) ∉ (coe : fin (n + 2) → ℕ) ⁻¹' (Iic n),
-    { simp only [mem_preimage, mem_Iic, not_le],
-      norm_cast,
-      rw fin.coe_coe_of_lt;
-      linarith, },
+    change linear_independent 𝕜 (f ∘ (coe : fin (n + 1) → ℕ)) at h₀,
+    have h₄ : (i : fin (n + 1)) ∉ (coe : fin (n + 1) → ℕ) ⁻¹' (Iio i),
+    { simp only [mem_preimage, mem_Iio, not_le],
+      rw [fin.coe_coe_of_lt, not_lt],
+      exact hi },
     apply linear_independent.not_mem_span_image h₀ h₄,
     rw [image_comp, image_preimage_eq_inter_range],
     simp only [function.comp_app, subtype.range_coe_subtype],
     convert h₃,
-    { norm_cast,
-      refine fin.coe_coe_of_lt (by linarith), },
-    { simp only [inter_eq_left_iff_subset, Iic, set_of_subset_set_of],
-      exact (λ a ha, by linarith), }, },
+    { exact fin.coe_coe_of_lt hi, },
+    { simp only [inter_eq_left_iff_subset, Iio, set_of_subset_set_of],
+      exact (λ a ha, by linarith) } }
 end
 
 /-- If the input of `gram_schmidt` is linearly independent, then the output is non-zero. -/
 lemma gram_schmidt_ne_zero' (f : ℕ → E) (h₀ : linear_independent 𝕜 f) (n : ℕ) :
-  gram_schmidt 𝕜 f n ≠ 0 :=
-gram_schmidt_ne_zero 𝕜 f n (linear_independent.comp h₀ _ (fin.coe_injective))
+  ∀ i, i < n → gram_schmidt 𝕜 f i ≠ 0 :=
+λ i hi, gram_schmidt_ne_zero 𝕜 f n (linear_independent.comp h₀ _ (fin.coe_injective)) i hi
```

Comment:
This generalization doesn't actually generalize anything, because you can always just pick `n = i.succ` and `n` doesn't appear in the goal.